fix: update available properties (#2157)
vaadin/docs: articles/configuration/properties.asciidoc

[NOTE]
If you use Spring Boot, you should add the prefix `vaadin.` (e.g., `vaadin.productionMode=true`).

[cols="1,4,1"]
|===
|Property name
|Description
|Default value

|`brotli`
|Checks whether pre-compressed Brotli files should be used if available.
|`true`

|`closeIdleSessions`

> |When set to `true`, the session is closed if no UI is active. From the servlet container's viewpoint, heartbeat requests are like any other request. This means that as long as there is an open UI, the session never expires, even if there is no user interaction. You can control this behavior by setting the init parameter named `closeIdleSessions` to `true`.

Author: Yes. Also the description is a hard to follow so I updated it a bit.

|`false`

|`devmode.liveReload.enabled`
|This means that if you are using some live reload tool on the server side, the browser is refreshed automatically after the code is reloaded on the server side.
|`true`

|`devmode.optimizeBundle`
|By default, in development mode all front-end resources found on the class path are included in the generated webpack bundle. When set to `true`, this creates an optimized bundle by including only front-end resources that are used from the application entry points. It uses bytecode scanning, which increases application start-up time.
|`false` for development mode, `true` for production mode

|`devmode.sessionSerialization.enabled`

> |This determines whether the [classname]`VaadinSession` instances of the users are serialized on server shutdown when running in development mode. When the parameter is set to `false`, only other HTTP session data is serialized or deserialized on development server restart. Enabling the property allows, for example, access control information to be preserved during development, so that you don't need to log in again for each change.

Author: Updated description

|5000 ms

|`pnpm.enable`
|This flag can be used to enable `pnpm` instead of `npm` to resolve and download front-end dependencies. By default, this flag is set to `false`, and `npm` is used. Setting it to `true` enables `pnpm`. See how to <<npm-pnpm#, switch between npm and pnpm>>.
|`false`

|`productionMode`
|This sets the application to work in production mode. Production mode disables most of the logged information that appears on the console. Logging and other debugging features can have a significant impact on performance otherwise. Development-mode JavaScript functions aren't exported. A `push` is given as a minified JavaScript file instead of a full size one, and static resources are cached. See <<../production#,Deploying to Production>> for more information.
|`false`

|`pushLongPollingSuspendTimeout`
|When using the long polling transport strategy, this specifies how long it accepts responses after each network request, in milliseconds.
|`-1` (no timeout)

|`pushMode`

> |The permitted values are "disabled" or "manual". See <<{articles}/advanced/server-push#, Server Push>> for more information.

Author: Added.

|`disabled`

|`pushServletMapping`

> |Returns the servlet mapping that bidirectional ("push") client-server communication should use.

Reviewer: Sorry, missed that one on the previous review
```suggestion
|Sets the servlet mapping that bidirectional ("push") client-server communication should use.
```

|`""`

|`requestTiming`
|If this is set to `true`, the server includes some basic timing information in each response. This can be used for performance testing.
|`true` for development mode, `false` for production mode

|`sendUrlsAsParameters`
|Returns `true` if the sending of URLs as GET and POST parameters in requests with content-type `application/x-www-form-urlencoded` is enabled.
|`true`

|`syncIdCheck`

> |Returns whether sync ID checking is enabled. The sync ID is used to handle situations in which the client sends a message to a connector that has been removed recently from the server.

Reviewer: ```suggestion
|Enables or disables the sync ID checking. The sync ID is used to handle situations in which the client sends a message to a connector that has been removed recently from the server.
```

|`true`

|`useDeprecatedV14Bootstrapping`
|This flag can be used to enable the server-side bootstrapping mode, which was used in Vaadin 14 and earlier versions. This option is only supported if webpack is used as the front-end build tool and not if the application uses Vite, which is the default. You can <<{articles}/configuration/live-reload#webpack-feature-flag,enable webpack using its associated feature flag>>.
|`false` (mode removed in v24)

|`webComponentDisconnect`

> |Returns the number of seconds that a WebComponent will wait for a reconnect before removing the server-side component from memory.

Reviewer: ```suggestion
|Defines the number of seconds that a WebComponent will wait for a reconnect before removing the server-side component from memory.
```
